### PR TITLE
Updatd development contribution docs

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Updated the documentation on how to develop HyperShift components in-cluster

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [ ] Relevant issues have been referenced.
- [x] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: docs/content/contribute/develop-in-cluster.md [hunk 1]
```diff
@@ -135,6 +138,27 @@ oc debug --namespace hypershift deployments/operator --image $(hack/publish-ocp.
 Your latest code should be deployed and logs should soon begin streaming. Just
 press `ctrl-c` to terminate and delete the pod.
 
+## Build and Launch a custom `hypershift-operator` image using a public registry
```
Comment (bryan-cox) [inline docs/content/contribute/develop-in-cluster.md:141]: Is this the same content already available on this [page](https://hypershift-docs.netlify.app/contribute/custom-images)? Maybe some of this content would be better added there?

Maybe a link could be added here on this page as a reference to that page instead.

Comment (muraee) [reply docs/content/contribute/develop-in-cluster.md]: This page is not accessible with knowing the link. I added a link to it.

**Comment by enxebre** [comment]
thanks!
/appprove

**Comment by sjenning** [comment]
approval didn't take before
/approve

**Pull request merged**